balance_service/utils/openstack/keystone.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import requests
+
+from django.conf import settings
+
+from balance_service import exceptions
+
+LOG = logging.getLogger(__name__)
+
+
+class KeystoneAPI:
+    def __init__(self, token, auth_url=None):
+        if not auth_url:
+            self.auth_url = settings.OPENSTACK_TACC_AUTH_URL
```

Comment:
Err, I don't think that's a good idea. I think auth_url should be required and an error raised if it's None. I don't think there's a sane thing we can fall back to. There is no guarantee the user will exist at TACC anymore, if they never logged in there.